contingency/contingency-api/src/main/java/com/powsybl/contingency/Contingency.java

```diff
@@ -30,12 +30,25 @@ public class Contingency extends AbstractExtendable<Contingency> {
 
     private final List<ContingencyElement> elements;
 
+    private final String name;
+
     public Contingency(String id, ContingencyElement... elements) {
         this(id, Arrays.asList(elements));
     }
 
     public Contingency(String id, List<ContingencyElement> elements) {
         this.id = Objects.requireNonNull(id);
+        this.name = null;
+        this.elements = new ArrayList<>(Objects.requireNonNull(elements));
+    }
+
+    public Contingency(String id, String name, ContingencyElement... elements) {
+        this(id, name, Arrays.asList(elements));
+    }
+
+    public Contingency(String id, String name, List<ContingencyElement> elements) {
+        this.id = Objects.requireNonNull(id);
+        this.name = Objects.requireNonNull(name);
```

Comment:
The use of requireNonNull seems a bit strange here because name can be null in a instance. Would it not be simpler to just drop the requireNonNull and use this constructor at line 40 ?
You can then drop the test in ContingencyDeserializer at line 63.

Reply:
ok